### ISSUE TITLE
FreeSWITCH listening interface and WSS → WS

### ISSUE THREAD

**Issue by lkiesow** [bigbluebutton/bigbluebutton#10968]
To fix issues with FreeSWITCH deadlocking, the documentation recently [recommends switching the internal communication between Nginx and FreeSWITCH from WSS to WS](https://github.com/bigbluebutton/bigbluebutton.github.io/commit/3cd09d6af1022de67ec9cc68e93f528d26be2cbf#diff-369390849b8a29e9b36bd8460a7b0215b433cbfe6be7890ac189d111ad44ba59):

> Note: As of BigBlueButton 2.2.29, you can have nginx proxy incoming WSS connections to WS on port 5066 to work around a [bug in FreeSWITCH](https://github.com/bigbluebutton/bigbluebutton/issues/9667).  If you are using 2.2.29 or later, you can run the following commands to implement the workaround:
>
> ~~~
> yq w -i /usr/share/meteor/bundle/programs/server/assets/app/config/settings.yml public.media.sipjsHackViaWs true
> sed -i 's/https/http/g'  /etc/bigbluebutton/nginx/sip.nginx 
> sed -i 's/7443/5066/g'  /etc/bigbluebutton/nginx/sip.nginx 
> ~~~

This basically undoes what was configured directly above this block (it instructs you to switch from WS to WSS).

Since we usually terminate TLS with Nginx and use non-secured connections internally (which should be just fine), the question is if we wouldn't want to just let Nginx terminate TLS of the WS connection by default (so users can't accidentally mess that up) and also make FreeSWITCH listen to localhost only.

In other words, does it make sense to change the default configuration to:

---

`/etc/bigbluebutton/nginx/sip.nginx`:
```
location /ws {
    proxy_pass http://127.0.0.1:5066;
    ...
}
```

---

`/opt/freeswitch/conf/sip_profiles/external.xml`:
```xml
  ...
  <param name="ws-binding" value="127.0.0.1:5066"/>
  ...
```

---

`/usr/share/meteor/bundle/programs/server/assets/app/config/settings.yml`:
```yml
  ...
  sipjsHackViaWs: true
  ...
```
---
This configuration works just fine and has the benefit of FreeSWITCH not listening on the public interface for this port. In assition to that, we can then also drop (or comment out) the WSS connection of FreeSWITCH.

**Comment by aguerson** [comment]
+1

**Comment by prlanzarin** [comment]
AFAIK it's in the pipeline to make it the default setting already. [1]
Also setting it to 127.0.0.1 might have unintended side effects, so not a good idea without very thorough testing beforehand. That's why it's recommended to be done the way 7443 worked. [2]

[1] https://github.com/bigbluebutton/bigbluebutton/issues/9667#issuecomment-735292999
[2] https://github.com/bigbluebutton/bigbluebutton/issues/9667#issuecomment-727598755

I am also thinking this issue might be considered sort of a duplicate of 9667.

**Comment by basisbit** [comment]
I agree to what @prlanzarin wrote. This discussion should be part of issue 9667 because it only discusses details of the problem and possible solution that was discussed in the mentioned issue. No need for a second discussion about the same problem.

**Comment by aguerson** [comment]
@lkiesow
I tried to replace with 127.0.0.1.
For the moment it is OK.
I will try more on Monday...

**Comment by aguerson** [comment]
```
# Warning: The setting of 127.0.0.1 for proxy_pass in
#
#    /etc/bigbluebutton/nginx/sip.nginx
#
# does not match the local IP address (x.x.x.x).
# (This is OK if you've manually changed the values)


#
# Kurento is will try to connect to x.x.x.x but FreeSWITCH is listening on 127.0.0.1 for port 5066
#
# To fix, run the commands
#
# sudo yq w -i /usr/local/bigbluebutton/bbb-webrtc-sfu/config/default.yml freeswitch.sip_ip 127.0.0.1
# sudo chown bigbluebutton:bigbluebutton /usr/local/bigbluebutton/bbb-webrtc-sfu/config/default.yml
#
```

**Comment by 2954456878** [comment]
Has it been solved in this way

**Comment by 2954456878** [comment]
Has it been solved in this way？

**Comment by basisbit** [comment]
@2954456878 this issue has been closed and is linked to #9667, as mentioned above. Please see #9667 for any details.
